### PR TITLE
Add support for batching (take two)

### PR DESCRIPTION
**Pull request by lantiga**
This PR reopens #241, which was merged and then reverted.

Issues not covered by tests emerged and need to be resolved before we merge again.
Details will follow here.

### CODE REVIEW

**Comment by lantiga** [comment]
Todo:
- [ ] Valgrind run (see also #274)
- [ ] Diagnose and fix crash report (block `MODELRUN`, kill client, next request -> crash)

**Comment by lantiga** [comment]
Valgrind looks legit, I can't reproduce the crash anymore.
@filipecosta90 ready to review and merge

**Comment by lantiga** [comment]
@filipecosta90 @hhsecond ready for review/approval

**Comment by lantiga** [comment]
I fixed a last (embarassing) bug in queue eviction (#310).
@hhsecond @filipecosta90 I'll be waiting for a last check before merging, let's try to make it happen as soon as we can.

File: src/backends/onnxruntime.c [hunk 1]
```diff
@@ -78,32 +78,82 @@ DLDataType RAI_GetDLDataTypeFromORT(ONNXTensorElementDataType dtype) {
...
+
+  if (count == 0) {
+    return NULL;
+  }
+
+  OrtAllocator *allocator;
+  status = ort->GetAllocatorWithDefaultOptions(&allocator);
   if (status != NULL) {
-    goto error;
+    return NULL;
```
Comment (tnachen) [inline src/backends/onnxruntime.c:92]: Is there a reason returning a status is ignored but line 437 it isn't?

**Pull request merged**